flopy/pakbase.py

```diff
@@ -243,8 +247,25 @@ def _check_flowp(self, f=None, verbose=True, level=1, checktype=None):
         # convert vertical anisotropy to Kv for checking
         if vka is not None:
             if "layvka" in self.__dict__:
-                for l in range(vka.shape[0]):
-                    vka[l] *= hk[l] if self.layvka.array[l] != 0 else 1
+                if self.parent.structured:
+                    for l in range(vka.shape[0]):
+                        vka[l] *= hk[l] if self.layvka.array[l] != 0 else 1
+                else:
+                    k_inds_to = np.cumsum(self.parent.disu.nodelay.array)
+                    k_inds_from = np.array(
+                        [
+                            k_inds_to[i - 1] if i > 0 else 0
+                            for i, n in enumerate(k_inds_to)
+                        ]
+                    )
+                    slices = [
+                        np.s_[k_inds_to[i - 1] : x] if i > 0 else np.s_[:x]
+                        for i, x in enumerate(k_inds_to)
+                    ]
+                    for k, sl in enumerate(slices):
+                        if self.layvka.array[k] != 0:
+                            vka[sl] *= hk[sl]
```

Comment:
And here, too, is some very unstructured grid specific code that is probably doesn't belong here.